examples/cpp_api/reading_sparse_layouts.cc

```diff
@@ -98,9 +98,8 @@ void read_array(tiledb_layout_t layout) {
   const std::vector<int> subarray = {1, 2, 2, 4};
 
   // Prepare buffers that will hold the results
-  auto max_el = array.max_buffer_elements(subarray);
-  std::vector<int> data(max_el["a"].second);
-  std::vector<int> coords(max_el[TILEDB_COORDS].second);
+  std::vector<int> data(6);
+  std::vector<int> coords(12);
```

Comment:
Change this (also below) to work with separate buffers per dimension. Same for all examples (they could go in a single PR).

Reply:
I'll handle refactoring the coords in a separate PR, I changed some of them here just to work around that set_coordinates bug.